plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json

```
@@ -40,7 +40,11 @@
 	"textdomain": "woocommerce",
 	"usesContext": [ "postId", "query", "queryId" ],
 	"viewScript": "wc-product-filters-frontend",
-	"example": {},
+	"example": {
+		"attributes": {
+			"isPreview": true
+		}
```

Initially I was going to suggest using `innerBlocks` definition here ( like below ), but noticed that doesn't quite work out of the box and still requires an explicit render of `InnerBlocks`. Even than I noticed that the translation is not working correctly like it does in core with using `"__( 'Filters', 'woocommerce' )"`.
```suggestion
		"innerBlocks": [
			{
				"name": "core/heading",
				"attributes": {
					"level": 3,
					"content": "__( 'Filters', 'woocommerce' )",
					"style": {
						"margin": { "top": "0", "bottom": "0" },
						"spacing": { "margin": { "top": "0", "bottom": "0" } }
					}
				}
			},
			{
				"name": "woocommerce/product-filter-active"
			},
			{
				"name": "woocommerce/product-filter-price"
			},
			{
				"name": "woocommerce/product-filter-rating"
			},
			{
				"name": "woocommerce/product-filter-attribute"
			},
			{
				"name": "woocommerce/product-filter-status"
			}
		]
```

---

Did some further digging and it did seem like the preview didn't work because of some CSS that the `wc-block-product-filters__overlay` classes added, like the `transform` call in combination with `absolute`: https://github.com/woocommerce/woocommerce/blob/855111f0620e45858ba2760e646afc463c4c1ab4/plugins/woocommerce-blocks/assets/js/blocks/product-filters/style.scss#L52

I am still not opposed to using this approach given the translation wouldn't work.

---

Thanks for the review. Yeah this issue was a bit tricky because this block is a mobile first approach. That's why you see that CSS transform. I also tried to do a CSS only approach but it didn't work the way I wanted. Is there any other concerns with the approach I have here?

---

> Btw in testing this, ( not a blocker of this PR ) I noticed a weird issue when using the filters in the page editor, not sure if you are aware:

There shouldn't be any interactions on the checkbox. You're able to click on the checkbox?? What theme are you using?

---

> Thanks for the review. Yeah this issue was a bit tricky because this block is a mobile first approach. That's why you see that CSS transform. I also tried to do a CSS only approach but it didn't work the way I wanted. Is there any other concerns with the approach I have here?

No concerns, just wanted to highlight it, and sounds like you already went down the same rabbit hole meaning I am happy with this approach as it seems like the best alternative IMO.

> There shouldn't be any interactions on the checkbox. You're able to click on the checkbox?? What theme are you using?

I didn't try clicking actually. I just tried it and that part does work correctly ( no interactions ). Just the oddities in the styling of the checkboxes. I was using theme 2024, but now doing some further testing it looks like I had a conflict with my custom `mu-plugin`. I removed the mu-plugin and things look good again 👍